### PR TITLE
ICU-21123 Fix ICU4J following ldml2icu change (and update JAR files)

### PR DESCRIPTION
**Pull request by hugovdm**
This one must wait until after pedberg's 1551!

##### Checklist

- [X] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21123
- [X] Updated PR title and link in previous line to include Issue number
- [X] Issue accepted
- [X] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java [hunk 1]
```diff
@@ -70,7 +73,12 @@ public PluralTableSink(String[] outArray) {
         public void put(UResource.Key key, UResource.Value value, boolean noFallback) {
             UResource.Table pluralsTable = value.getTable();
             for (int i = 0; pluralsTable.getKeyAndValue(i, key, value); ++i) {
-                int index = getIndex(key.toString());
+                String keyString = key.toString();
+                // TODO(ICU-21123): Load the correct inflected form, possibly from the "case" structure.
+                if (keyString.equals("case")) {
+                    continue;
+                }
```
Comment (hugovdm) [inline icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java:80]: Busy working on case handling and making use of gender. This PR now gets ICU4J in sync with what's in master for ICU4C.

**Review by sffc — APPROVED** [review]
RSLGTM

File: icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java (hunk 1 above)
Comment (younies) [reply icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java:80]: +1

File: icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java [hunk 2]
```diff
@@ -71,8 +74,8 @@ public void put(UResource.Key key, UResource.Value value, boolean noFallback) {
             UResource.Table pluralsTable = value.getTable();
             for (int i = 0; pluralsTable.getKeyAndValue(i, key, value); ++i) {
                 String keyString = key.toString();
-                if (keyString.equals("case") || keyString.equals("gender")) {
-                    // TODO: @Hugo to fix for new grammatical stuff
+                // TODO(ICU-21123): Load the correct inflected form, possibly from the "case" structure.
+                if (keyString.equals("case")) {
```
Comment (younies) [inline icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java:78]: So, you just putting the skeleton and the implementation in another PR

**Review by younies — APPROVED** [review]

File: icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java [hunk 3]
```diff
@@ -29,14 +29,17 @@
 
     private static final int DNAM_INDEX = StandardPlural.COUNT;
     private static final int PER_INDEX = StandardPlural.COUNT + 1;
-    static final int ARRAY_LENGTH = StandardPlural.COUNT + 2;
+    private static final int GENDER_INDEX = StandardPlural.COUNT + 2;
+    static final int ARRAY_LENGTH = StandardPlural.COUNT + 3;
 
     private static int getIndex(String pluralKeyword) {
         // pluralKeyword can also be "dnam" or "per"
```
Comment (younies) [inline icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java:36]: `        // pluralKeyword can also be "dnam", "per" or "gender"`

**Comment by hugovdm** [comment]
Part of Younies' ICU4J PR: #1590.

**Pull request closed without merging**